front/src/modules/rollingStock/helpers/__tests__/utils.spec.ts

```diff
@@ -187,3 +201,224 @@ describe('checkRollingStockFormValidity', () => {
+    });
+  });
+  describe('mass converter', () => {
+    it('should divide the unit value by mass with current mass in t', () => {
+      const convertedValue = convertUnitsWithMass('kN', 'kN/t', 40, 't', 20);
+      expect(convertedValue).toEqual(0.5);
+    });
+    it('should divide the unit value by mass with current mass in kg', () => {
+      const convertedValue = convertUnitsWithMass('kN', 'kN/t', 40000, 'kg', 20);
+      expect(convertedValue).toEqual(0.5);
+    });
+    it('should multiply the unit value by mass with current mass in t', () => {
+      const convertedValue = convertUnitsWithMass('kN/t', 'kN', 40, 't', 0.5);
+      expect(convertedValue).toEqual(20);
+    });
+    it('should multiply the unit value by mass with current mass in kg', () => {
+      const convertedValue = convertUnitsWithMass('kN/t', 'kN', 40000, 'kg', 0.5);
+      expect(convertedValue).toEqual(20);
+    });
+  });
+  describe('parameter unit convertor', () => {
+    const options: InputGroupSNCFValue<MultiUnit>[] = [
+      {
+        unit: 'kg',
+        value: undefined,
+      },
+      {
+        unit: 'km/h',
+        value: 200,
+      },
+      {
+        unit: 'N/(km/h)',
+        value: 0,
+      },
+      {
+        unit: 'kN/t',
+        value: 15000,
+      },
+    ];
+    const multiUnitsParams: Record<string, MultiUnitsParameter> = {
+      mass: {
+        min: 0.1,
+        max: 10000,
+        unit: 't',
+        value: 100,
+      },
+      maxSpeed: {
+        min: 1,
+        max: 600,
+        unit: 'km/h',
+        value: 200,
+      },
+      rollingResistanceB: {
+        min: 0,
+        max: 1800,
+        unit: 'N/(m/s)',
+        value: 0,
+      },
+      rollingResistanceA: {
+        min: 0,
+        max: 20000,
+        unit: 'N',
+        value: 15000,
+      },
+    };
+    it("should return undefined if it converts a param value and the option's value is undefined", () => {
+      const convertedValue = handleUnitValue(
+        options[0],
+        multiUnitsParams.mass,
+        multiUnitsParams.mass
+      );
+      expect(convertedValue).toEqual(undefined);
+    });
+    it("should return the same option value if option's unit is the same as the param's current unit", () => {
+      const convertedMinValue = handleUnitValue(
+        options[1],
+        multiUnitsParams.maxSpeed,
+        multiUnitsParams.mass,
+        'min'
+      );
+      expect(convertedMinValue).toEqual(convertedMinValue);
+    });
+    it("should return the same option value if option's unit equals 0", () => {
+      const convertedValue = handleUnitValue(
+        options[2],
+        multiUnitsParams.rollingResistanceB,
+        multiUnitsParams.mass
+      );
+      expect(convertedValue).toEqual(convertedValue);
+    });
+    it("should properly convert the unit's value if units are different and there is an option value", () => {
+      const convertedMinValue = handleUnitValue(
+        options[0],
```

Comment:
Maybe it's better like this
```suggestion
      {
        unit: 'kg',
        value: undefined,
      },
```

Reply:
Yep I corrected it on the values const